### PR TITLE
handle ptf packages inside of normal pkg.remove function

### PR DESCRIPTION
**Pull request by mcalmer**
### What issues does this PR fix or reference?

Enhance zypper pkg module to remove PTF (Product Temporary Fix) packages.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/zypperpkg.py [hunk 1]
```diff
@@ -2081,6 +2092,10 @@ def remove(
         salt '*' pkg.remove <package name>
         salt '*' pkg.remove <package1>,<package2>,<package3>
         salt '*' pkg.remove pkgs='["foo", "bar"]'
+
+    .. versionadded:: 3007
```
Comment (Ch3LL) [inline salt/modules/zypperpkg.py:2096]:
```suggestion
    .. versionchanged:: 3007
        Can now remove also PTF packages which require a different handling in the backend.
```

File: salt/modules/zypperpkg.py [hunk 2]
```diff
@@ -2548,6 +2563,22 @@ def _get_visible_patterns(root=None):
...
+    """
+    Find ptf packages in "pkgs" and return them as list
+    """
+    ptfs = []
+    cmd = ["rpm"]
+    if root:
+        cmd.extend(["--root", root])
+    cmd.extend(["-q", "--provides"])
+    for pkg in pkgs:
+        output = __salt__["cmd.run"](cmd + [pkg])
```
Comment (Ch3LL) [inline salt/modules/zypperpkg.py:2576]:
```
[root@7635debe6eea /]# rpm -q --provides grep lua     
bundled(gnulib)
config(grep) = 2.20-3.el7
grep = 2.20-3.el7
grep(x86-64) = 2.20-3.el7
liblua-5.1.so()(64bit)
lua = 5.1
lua = 5.1.4-15.el7
lua(abi) = 5.1
lua(x86-64) = 5.1.4-15.el7
```

It looks like you can pass multiple packages to that command. Is it possible to only run this command once with all of the pkgs passed in to decrease the amount of commands we need to run?

**Review by Ch3LL — COMMENTED** [review]
I like this approach much better. Thanks :) I just left a couple more suggestions.

File: salt/modules/zypperpkg.py (hunk 2 above)
Comment (mcalmer) [reply salt/modules/zypperpkg.py:2577]: The problem is, I need to know which output belong to which package in the input line.
I will play a bit around with query format and see if we can do something about it.

Comment (mcalmer) [reply salt/modules/zypperpkg.py]: I found a solution and implemented it

**Review by Ch3LL — DISMISSED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**